snuba/datasets/transactions_processor.py

```diff
@@ -0,0 +1,115 @@
+        processed = {'deleted': 0}
+        if not isinstance(message, (list, tuple)) and len(message) >= 2:
+            return None
+        version = message[0]
+        if version not in (0, 1, 2):
+            return None
+        type_, event = message[1:3]
+        if type_ != 'insert':
+            return None
+
+        data = event.get("data", {})
+        event_type = data.get("type")
+        if event_type != "transaction":
+            return None
+        base = extract_base(event)
+        enforce_retention(
+            event,
+            datetime.fromtimestamp(data['timestamp']),
+            base['retention_days'],
+            base['project_id'],
+        )
+        processed.update(base)
+
+        transaction_ctx = data["contexts"]["trace"]
+        trace_id = transaction_ctx["trace_id"]
+        processed["event_id"] = str(uuid.UUID(processed["event_id"]))
+        processed["trace_id"] = str(uuid.UUID(trace_id))
+        processed["span_id"] = int(transaction_ctx["span_id"], 16)
+        processed["transaction_op"] = _unicodify(transaction_ctx.get("op", ""))
+        processed["transaction_name"] = _unicodify(data["transaction"])
+
+        processed["start_ts"], processed["start_ms"] = self.__extract_timestamp(
+            data["start_timestamp"],
+        )
+        processed["finish_ts"], processed["finish_ms"] = self.__extract_timestamp(
+            data["timestamp"],
+        )
+
+        processed['platform'] = _unicodify(event['platform'])
+
+        tags = _as_dict_safe(data.get('tags', None))
+        extract_extra_tags(processed, tags)
+
+        promoted_tag = {col: _unicodify(tags[col])
+            for col in self.PROMOTED_TAGS
+            if col in tags
+        }
+        processed["release"] = promoted_tag.get(
+            "sentry:release",
+            event.get("release"),
+        )
+        processed["environment"] = promoted_tag.get("environment")
+
+        contexts = _as_dict_safe(data.get('contexts', None))
+        extract_extra_contexts(processed, contexts)
+
+        processed["dist"] = _unicodify(
+            promoted_tag.get("sentry:dist",
+            data.get("dist", "BA")),
+        )
+
+        user_data = {}
+        extract_user(user_data, data.get("user", {}))
+        processed["user"] = promoted_tag.get("sentry:user", "")
+        processed["user_name"] = user_data.get("username")
+        processed["user_id"] = user_data.get("user_id")
+        processed["user_email"] = user_data.get("email")
+        ip_address = user_data.get("ip_address")
+        if ip_address:
+            ip_address = ipaddress.ip_address(ip_address)
+            if ip_address.version == 4:
+                processed["ip_address_v4"] = str(ip_address)
+            elif ip_address.version == 6:
+                processed["ip_address_v6"] = str(ip_address)
+
+        if metadata is not None:
+            processed['offset'] = metadata.offset
+            processed['partition'] = metadata.partition
```

Comment:
Same nit here as schema re: ordering of these columns

Reply:
done